### ISSUE TITLE
Game crashes when using `InputEvent.isPressed()`

### ISSUE THREAD

**Issue by JustinSharkey** [migueldeicaza/SwiftGodot#35]
Setup: I have subclassed `Node` and overridden the `_input(event:)` function.

When using `InputEvent.isPressed()` or `InputEvent.isEcho()` the Game crashes.

<img width="541" alt="Screenshot 2023-04-28 at 4 46 15 PM" src="https://user-images.githubusercontent.com/3056529/235251275-bdbdb6f7-1c41-491b-bb4b-946f9bd3bfdf.png">

<img width="1164" alt="Screenshot 2023-04-28 at 10 02 09 AM" src="https://user-images.githubusercontent.com/3056529/235251080-33729485-d12c-4d93-9ce5-f3a09f9adf4f.png">

[Godot-2023-04-28-100427.ips.zip](https://github.com/migueldeicaza/SwiftGodot/files/11357226/Godot-2023-04-28-100427.ips.zip)

**Comment by migueldeicaza** [comment]
Thanks for the bug report!  Will investigate!

**Comment by migueldeicaza** [comment]
Still working on this.

I figured out what is going on.

The raw pointers that I get out on the callback are not object pointers, they are some kind of wrapper.   I say "some kind" because the source code indicates that these are Variants(), but they do not look like variants to me.   When I looked at the C++ binding, it looks like I need to call the ref_get_object/object_get_instance_binding pair on those.

The only complexity now is that the latter requires a set of management callbacks (which I already have), but the odd part is that the way this is written will make it hard to decide if what I am getting back is a Framework type or a User Type (the callbacks I have), so it is not clear how to properly unwrap this.
